src/MediaWiki.ts

```diff
@@ -11,26 +11,126 @@ import basicURLDirector from './util/builders/url/basic.director.js'
   private apiUrlDirector: ApiURLDirector
+  private baseUrlDirector: BaseURLDirector
+
+  private _veapiUrl: URL
+  private _restApiUrl: URL
+  private _apiUrl: URL
+  private _modulePath: string
+  private _webUrl: URL
+  private _desktopRestApiUrl: URL
+  // TODO: Mobile builder was removed since there is no /mobile-sections endpoint
+
+  // Set default MW capabilities
+  private readonly mwCapabilities: MWCapabilities
+
+  /**
+   * veApiUrl based on top of 'new ApiURLDirecto'
+   */
+  public get veapiUrl(): URL {
+    if (!this._veapiUrl) {
+      // TODO: This depend on baseUrlDirector.buildURL(this.apiPath) and looks like a weak solution
+      this._veapiUrl = this.apiUrlDirector.buildVisualEditorURL()
+    }
+    return this._veapiUrl
+  }
+
+  /**
+   * restApiUrl, apiUrl, modulePath, webUrl and desktopRestApiUr are based on top of 'new BaseURLDirector'
+   */
+  public get restApiUrl(): URL {
+    if (!this._restApiUrl) {
+      this._restApiUrl = this.baseUrlDirector.buildRestApiURL(this.restApiPath)
+    }
+    // TODO: define usage of this property
+    return this._restApiUrl
+  }
+
+  public get apiUrl(): URL {
+    if (!this._apiUrl) {
+      this._apiUrl = this.baseUrlDirector.buildURL(this.apiPath)
+    }
+    return this._apiUrl
+  }
+
+  public get modulePath() {
+    if (!this._modulePath) {
+      this._modulePath = this.baseUrlDirector.buildModuleURL(this.modulePathConfig)
+    }
+    return this._modulePath
+  }
+
+  public get webUrl(): URL {
+    if (!this._webUrl) {
+      this._webUrl = this.baseUrlDirector.buildURL(this.wikiPath)
+    }
+    return this._webUrl
+  }
+
+  public get desktopRestApiUrl(): URL {
+    if (!this._desktopRestApiUrl) {
+      this._desktopRestApiUrl = this.baseUrlDirector.buildDesktopRestApiURL(this.restApiPath)
+    }
+    return this._desktopRestApiUrl
+  }
+
+  public hasDesktopRestApi = async function (loginCookie?: string, testArticleId?: string): Promise<any> {
+    const desktopRestApiAvailable = await this.checkApiAvailabilty(this.getDesktopRestApiArticleUrl(testArticleId), loginCookie)
+    this.hasDesktopRestApi = async function (): Promise<boolean> {
+      return desktopRestApiAvailable
+    }
+  }
+
+  public hasVeApi = async function (loginCookie?: string, testArticleId?: string): Promise<any> {
+    const veRestApiAvailable = await this.checkApiAvailabilty(this.getVeApiArticleUrl(testArticleId), loginCookie)
+    this.hasVeApi = async function (): Promise<boolean> {
+      return veRestApiAvailable
+    }
+  }
+
+  public hasCoordinatesApi = async function (downloader?: Downloader): Promise<any> {
+    const validNamespaceIds = this.namespacesToMirror.map((ns) => this.namespaces[ns].num)
+    const reqOpts = {
+      action: 'query',
+      format: 'json',
+      // TODO: Do we need this.mwCapabilities.coordinatesAvailable here?
+      prop: `redirects|revisions${this.mwCapabilities.coordinatesAvailable ? '|coordinates' : ''}${this.getCategories ? '|categories' : ''}`,
```

Comment:
Here there is a chicken-egg problem, not sure how this happened, but you clearly need to put `|coordinates` to then see if this is supported.